test_runner/src/main/kotlin/ftl/filter/TestFilters.kt

```diff
@@ -0,0 +1,89 @@
+package ftl.filter
+
+import com.linkedin.dex.parser.TestMethod
+import java.io.IOException
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.Locale
+import java.util.regex.Pattern
+
+typealias TestFilter = (TestMethod) -> Boolean
+
+object TestFilters {
+    private const val ARG_CLASS = "class"
+    private const val ARG_PACKAGE = "package"
+    private const val ARG_ANNOTATION = "annotation"
+    private const val ARG_TEST_FILE = "testfile"
```

Comment:
I think then it is better to ignore `testfile` for now. How do you think it would be possible without client side validation since we need to know exactly which methods to run?

Reply:
In the documentation it is mentioned that 
> Any test filter supported by `am instrument -e ...` is supported.

According to the [AndroidJUnitRunner JavaDoc](https://developer.android.com/reference/android/support/test/runner/AndroidJUnitRunner) that would mean:
```
public static final String ARGUMENT_TEST_CLASS = "class";
private static final String ARGUMENT_TEST_SIZE = "size";
private static final String ARGUMENT_ANNOTATION = "annotation";
private static final String ARGUMENT_NOT_ANNOTATION = "notAnnotation";
private static final String ARGUMENT_TEST_PACKAGE = "package";
private static final String ARGUMENT_TEST_FILE = "testFile";
```

Reply:
I think that's a great idea. We'd probably want to make sure we're using the latest alpha code. Last time I checked, that source was not up to date with the AndroidX espresso.

Reply:
re: testFile, that's my mistake. You're correct this is a supported feature. testFile under test-targets makes sense.